### PR TITLE
RTShaderSystem: range loop refactoring

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

File: Components/RTShaderSystem/src/OgreShaderGenerator.cpp [hunk 1]
```diff
@@ -965,13 +965,9 @@ bool ShaderGenerator::cloneShaderBasedTechniques(Material& srcMat, Material& dst
                         RenderState* srcRenderState = t->getRenderState(pi);
                         RenderState* dstRenderState = getRenderState(srcToTechniqueScheme, dstMat, pi);
 
-                        const SubRenderStateList& srcSubRenderState =
-                            srcRenderState->getSubRenderStates();
-
-                        SubRenderStateList::const_iterator itSubState = srcSubRenderState.begin(), itSubStateEnd = srcSubRenderState.end();
-                        for(;itSubState != itSubStateEnd ; ++itSubState)
+                        for(const auto& s : srcRenderState->getSubRenderStates())
```
Comment (paroj) [inline Components/RTShaderSystem/src/OgreShaderGenerator.cpp:968]:
```suggestion
                        for(SubRenderState* srcSubState : srcRenderState->getSubRenderStates())
```

Comment (Joilnen) [reply Components/RTShaderSystem/src/OgreShaderGenerator.cpp]: I did this 
for (SubRenderState* s : srcRenderState->getSubRenderStates())

File: Components/RTShaderSystem/src/OgreShaderGenerator.cpp [hunk 2]
```diff
@@ -965,13 +965,9 @@ bool ShaderGenerator::cloneShaderBasedTechniques(Material& srcMat, Material& dst
...
 
-                        const SubRenderStateList& srcSubRenderState =
-                            srcRenderState->getSubRenderStates();
-
-                        SubRenderStateList::const_iterator itSubState = srcSubRenderState.begin(), itSubStateEnd = srcSubRenderState.end();
-                        for(;itSubState != itSubStateEnd ; ++itSubState)
+                        for(SubRenderState* s : srcRenderState->getSubRenderStates())
                         {
-                            SubRenderState* srcSubState = *itSubState;
+                            SubRenderState* srcSubState = s;
```
Comment (paroj) [inline Components/RTShaderSystem/src/OgreShaderGenerator.cpp:970]:
```suggestion
```

**Comment by paroj** [comment]
please keep the dstSubState, srcSubState names

**Pull request merged**